pandas/_libs/interval.pyx

```diff
@@ -383,14 +383,14 @@ cdef class Interval(IntervalMixin):
         return f'{start_symbol}{left}, {right}{end_symbol}'
 
     def __add__(self, y):
-        if isinstance(y, numbers.Number):
+        if isinstance(y, (numbers.Number, Timedelta)):
             return Interval(self.left + y, self.right + y, closed=self.closed)
         elif isinstance(y, Interval) and isinstance(self, numbers.Number):
             return Interval(y.left + self, y.right + self, closed=y.closed)
         return NotImplemented
 
     def __sub__(self, y):
-        if isinstance(y, numbers.Number):
+        if isinstance(y, (numbers.Number, Timedelta)):
```

Comment:
in most contexts where we allow Timedelta we would also allow timedelta, np.timedelta64, and Tick.  Maybe something like:

```
if self_is_timedelta_or_timestamp:
     left = self.left.__sub__(y)
     if left is NotImplemented:
          return NotImplemented
     right = self.right.__sub__(y)
     return Interval(left, right, closed=self.closed)
else:
     ...
```

Reply:
that also would allow for adding/subtracting Timestamp, which we would probably want to allow in specific cases

Reply:
Interesting, could we also do that unconditionally, i.e., just always try it on `self.left` for any `self` / `y` (possibly letting it raise) and the continue from there? Also, the condition in the `elif` for `__add__` on master can never be true unless I'm missing something?

Reply:
i think that makes sense, but id like to wait for @jschendel to weigh in before going too far in this direction